DigitalLearningSolutions.Web/Attributes/SetApplicationTypeAttribute.cs

```diff
@@ -0,0 +1,56 @@
+﻿namespace DigitalLearningSolutions.Web.Attributes
+{
+    using System;
+    using DigitalLearningSolutions.Web.Models.Enums;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.AspNetCore.Mvc.Filters;
+
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method)]
+    public class SetApplicationTypeAttribute : Attribute, IActionFilter
+    {
+        private readonly string? applicationRouteParamName;
+        private readonly string? applicationTypeName;
+
+        public SetApplicationTypeAttribute(string? applicationTypeName = null, string? applicationRouteParamName = null)

```

Comment:
For some reason the names don't flow as nicely as I thought. Here's a suggestion, and if you don't think it's better let's stick with what we've got.
If we change `applicationRouteParamName` to `determiningRouteParameter`, is that any easier to understand?